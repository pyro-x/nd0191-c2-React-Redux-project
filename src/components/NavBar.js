import React from 'react';
import { Link } from 'react-router-dom';
import { useLocation ,useNavigate} from 'react-router-dom'
import { connect } from 'react-redux';
import { setAuthedUser } from '../actions/authedUser';


const NavBar = (props) => {
    console.log ("navbar props", props);
    const location = useLocation();
    const navigate = useNavigate();
    const {authedUser} = props;

    const handleLogout = () => {

        //props.dispatch({type: "LOGOUT"});
        props.dispatch(setAuthedUser(null));
        navigate("/login");
        
    }

    console.log ("location", location);
    return (
        <nav className='flex justify-between px-4 pt-4  pb-2 border-b-2 mt-2 shadow'>
            <ul className="flex space-x-2">

                <li className={location.pathname === "/" ? "border-b-4":""}>
                    <Link to="/">Home</Link>
                </li>
                <li className={location.pathname === "/leaderboard" ? "border-b-4":""}>
                    <Link to="/leaderboard">Leader board</Link>
                </li>
                <li className={location.pathname === "/add" ? "border-b-4":""}>
                    <Link to="/add">New</Link>
                </li>
            </ul>
            
               {
                authedUser !== null ? (  
                    <ul className="flex space-x-2 justify-items-end">

                        <li> 
                        </li>
                        <li>
                            <div className="flex">
                                <img src={authedUser.avatarURL} className='w-8 h-8 rounded-full mr-2' alt={authedUser.name}/>
                                {authedUser.name}
                            </div>
                        </li>
                        <li>
                            <div onClick={handleLogout}>Logout</div>
                        </li>
                    </ul>
                    ) : (
                        <ul className="flex space-x-2 justify-items-end">
                            <li>
                                <Link to="/login">Login</Link>
                            </li>
                        </ul>
                    )
                }
        </nav>
    );
}
const mapStateToProps = ({ users, authedUser }) => {
    return {
       authedUser: authedUser!== null ? users[authedUser] : null
    }
}

export default connect(mapStateToProps)(NavBar);