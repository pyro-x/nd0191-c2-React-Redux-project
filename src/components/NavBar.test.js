/* eslint-disable testing-library/prefer-screen-queries */

import * as React from 'react';
import {  render } from '@testing-library/react';

import reducer from "../reducers";
import { Provider } from 'react-redux';
import middleware from "../middleware";
import { createStore } from 'redux';
import { MemoryRouter } from 'react-router';

import NavBar from "./NavBar";


describe ('NavBar', () => {
    let store;
    const initialState = {
        authedUser: "sarahedo",
        users: {
            sarahedo: {
                id: "sarahedo",
                name: "Sarah Edo",
                avatarURL: "https://api.dicebear.com/5.x/avataaars/svg?seed=sarahedo&style=circle",
                answers: {
                    "8xf0y6ziyjabvozdd253nd": "optionOne",
                    "6ni6ok3ym7mf1p33lnez": "optionOne",
                    "am8ehyc8byjqgar0jgpub9": "optionTwo",
                    "loxhs1bqm25b708cmbf3g": "optionTwo"
                },
                questions: [],
                password: "password123"
            }
        }
    }

    beforeEach(() => {
        store = createStore(reducer, initialState, middleware);
    });

    it ('renders correctly', () => {
        const  view  = render(
            <MemoryRouter>
                <Provider store={store}>
                    <NavBar/>
                </Provider>
            </MemoryRouter>
            );
        expect(view).toMatchSnapshot()
    });

    it ('contains three main sections', () => {
        const  view  = render(
            <MemoryRouter>
                <Provider store={store}>
                    <NavBar/>
                </Provider>
            </MemoryRouter>
            );
        expect(view.getAllByRole('link')).toHaveLength(3);

        expect (view.getByTestId("home-lnk")).toBeInTheDocument();
        expect (view.getByTestId("add-lnk")).toBeInTheDocument();
        expect (view.getByTestId("leaderboard-lnk")).toBeInTheDocument();
    });

    it ('contains a logout button', () => {
        const  view  = render(
            <MemoryRouter>
                <Provider store={store}>
                    <NavBar/>
                </Provider>
            </MemoryRouter>
            );
        expect(view.getByTestId("logout-btn")).toBeInTheDocument();

    });

});

