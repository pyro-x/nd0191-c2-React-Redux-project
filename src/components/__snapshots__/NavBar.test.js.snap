// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="flex justify-between px-4 pt-4  pb-2 border-b-2 mt-2 shadow"
      >
        <ul
          class="flex space-x-2"
        >
          <li
            class="border-b-4"
          >
            <a
              data-testid="home-lnk"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class=""
          >
            <a
              data-testid="leaderboard-lnk"
              href="/leaderboard"
            >
              Leader board
            </a>
          </li>
          <li
            class=""
          >
            <a
              data-testid="add-lnk"
              href="/add"
            >
              New
            </a>
          </li>
        </ul>
        <ul
          class="flex space-x-2 justify-items-end"
        >
          <li />
          <li>
            <div
              class="flex"
            >
              <img
                alt="Sarah Edo"
                class="w-8 h-8 rounded-full mr-2"
                src="https://api.dicebear.com/5.x/avataaars/svg?seed=sarahedo&style=circle"
              />
              Sarah Edo
            </div>
          </li>
          <li>
            <div
              class="cursor-pointer"
              data-testid="logout-btn"
            >
              Logout
            </div>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="flex justify-between px-4 pt-4  pb-2 border-b-2 mt-2 shadow"
    >
      <ul
        class="flex space-x-2"
      >
        <li
          class="border-b-4"
        >
          <a
            data-testid="home-lnk"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class=""
        >
          <a
            data-testid="leaderboard-lnk"
            href="/leaderboard"
          >
            Leader board
          </a>
        </li>
        <li
          class=""
        >
          <a
            data-testid="add-lnk"
            href="/add"
          >
            New
          </a>
        </li>
      </ul>
      <ul
        class="flex space-x-2 justify-items-end"
      >
        <li />
        <li>
          <div
            class="flex"
          >
            <img
              alt="Sarah Edo"
              class="w-8 h-8 rounded-full mr-2"
              src="https://api.dicebear.com/5.x/avataaars/svg?seed=sarahedo&style=circle"
            />
            Sarah Edo
          </div>
        </li>
        <li>
          <div
            class="cursor-pointer"
            data-testid="logout-btn"
          >
            Logout
          </div>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
